rpc/core/tx.go

```diff
@@ -62,72 +70,78 @@ func (env *Environment) TxSearch(
 	orderBy string,
 ) (*ctypes.ResultTxSearch, error) {
 
-	// if index is disabled, return error
-	if _, ok := env.TxIndexer.(*null.TxIndex); ok {
-		return nil, errors.New("transaction indexing is disabled")
+	if !indexer.KVSinkEnabled(env.EventSinks) {
+		return nil, errors.New("transaction searching is disabled due to no kvEventSink")
 	}
 
 	q, err := tmquery.New(query)
 	if err != nil {
 		return nil, err
 	}
 
-	results, err := env.TxIndexer.Search(ctx.Context(), q)
-	if err != nil {
-		return nil, err
-	}
-
-	// sort results (must be done before pagination)
-	switch orderBy {
-	case "desc", "":
-		sort.Slice(results, func(i, j int) bool {
-			if results[i].Height == results[j].Height {
-				return results[i].Index > results[j].Index
+	for _, sink := range env.EventSinks {
+		if sink.Type() == indexer.KV {
+			results, err := sink.SearchTxEvents(ctx.Context(), q)
+			if err != nil {
+				return nil, err
 			}
-			return results[i].Height > results[j].Height
-		})
-	case "asc":
-		sort.Slice(results, func(i, j int) bool {
-			if results[i].Height == results[j].Height {
-				return results[i].Index < results[j].Index
-			}
-			return results[i].Height < results[j].Height
-		})
-	default:
-		return nil, fmt.Errorf("%w: expected order_by to be either `asc` or `desc` or empty", ctypes.ErrInvalidRequest)
-	}
 
-	// paginate results
-	totalCount := len(results)
-	perPage := env.validatePerPage(perPagePtr)
+			// sort results (must be done before pagination)
+			switch orderBy {
+			case "desc", "":
+				sort.Slice(results, func(i, j int) bool {
+					if results[i].Height == results[j].Height {
+						return results[i].Index > results[j].Index
+					}
+					return results[i].Height > results[j].Height
+				})
+			case "asc":
+				sort.Slice(results, func(i, j int) bool {
+					if results[i].Height == results[j].Height {
+						return results[i].Index < results[j].Index
+					}
+					return results[i].Height < results[j].Height
+				})
+			default:
+				return nil, fmt.Errorf("%w: expected order_by to be either `asc` or `desc` or empty", ctypes.ErrInvalidRequest)
```

Comment:
I see that this is legacy code, but I think that for the error wrapping to work %w has to be at the end.

Reply:
👍